uniffi_bindgen/src/bindings/kotlin/templates/NamespaceLibraryTemplate.kt

```diff
@@ -55,20 +55,65 @@ internal open class {{ ffi_struct.name()|ffi_struct_name }}(
 {%- endmatch %}
 {%- endfor %}
 
+
+{%- macro decl_kotlin_functions(func_list) -%}
+{% for func in func_list -%}
+fun {{ func.name() }}(
+    {%- call kt::arg_list_ffi_decl(func) %}
+): {% match func.return_type() %}{% when Some with (return_type) %}{{ return_type.borrow()|ffi_type_name_by_value }}{% when None %}Unit{% endmatch %}
+{% endfor %}
+{%- endmacro %}
+
+// For large crates we prevent `MethodTooLargeException` (see #2340)
+// N.B. the name of the extension is very misleading, since it is 
+// rather `InterfaceTooLargeException`, caused by too many methods 
+// in the interface for large crates.
+//
+// By splitting the otherwise huge interface into two parts
+// * UniffiLib 
+// * UniffiLibChecksums (this)
+// And all checksum methods are put into `UniffiLibChecksums`
+// we allow for ~2x as many methods in the UniffiLib interface.
+internal interface UniffiLibChecksums : Library {
+{%- call decl_kotlin_functions(ci.iter_checksum_ffi_functions()) %}
+}
+
 // A JNA Library to expose the extern-C FFI definitions.
 // This is an implementation detail which will be called internally by the public API.
-
 internal interface UniffiLib : Library {
     companion object {
         internal val INSTANCE: UniffiLib by lazy {
-            loadIndirect<UniffiLib>(componentName = "{{ ci.namespace() }}")
-            .also { lib: UniffiLib ->
-                uniffiCheckContractApiVersion(lib)
-                uniffiCheckApiChecksums(lib)
-                {% for fn in self.initialization_fns() -%}
-                {{ fn }}(lib)
-                {% endfor -%}
-            }
+            val componentName = "{{ ci.namespace() }}"
+            // For large crates we prevent `MethodTooLargeException` (see #2340)
+            // N.B. the name of the extension is very misleading, since it is 
+            // rather `InterfaceTooLargeException`, caused by too many methods 
+            // in the interface for large crates.
+            //
+            // By splitting the otherwise huge interface into two parts
+            // * UniffiLib (this)
+            // * UniffiLibChecksums
+            // And all checksum methods are put into `UniffiLibChecksums`
+            // we allow for ~2x as many methods in the UniffiLib interface.
+            // 
+            // Thus we first load the library with `loadIndirect` as `UniffiLibChecksums`
+            // so that we can call `uniffiCheckApiChecksums`...
+            loadIndirect<UniffiLibChecksums>(componentName)
+                .also { lib: UniffiLibChecksums ->
+                    uniffiCheckApiChecksums(lib)
```

Comment:
Let me know if I should adjust `interface UniffiLibChecksums` to contain the `ffi_uniffi_contract_version` function so that we can call `uniffiCheckContractApiVersion(lib)` inside this `.also` block, which would be closer to what we do on `main` branch (call `uniffiCheckContractApiVersion` **before** `uniffiCheckApiChecksums`).

Alternatively I can change to this:

```kotlin
val lib =  loadIndirect<UniffiLib>(componentName)
                .also { lib: UniffiLib ->
                   uniffiCheckContractApiVersion(lib)
                }

// check checksums
loadIndirect<UniffiLibChecksums>(componentName)
                .also { lib: UniffiLibChecksums ->
                    uniffiCheckApiChecksums(lib)

lib // return lib
```

to ensure `uniffiCheckContractApiVersion` is called before `uniffiCheckApiChecksums`?

Reply:
I prefer `2`, since a contract version change could potentially affect the checksum function signature.  However, if this makes the `interface` code more complicated, it's not worth it.

Reply:
I will give it a go

Reply:
@bendk OK I've renamed to new lib `IntegrityCheckingUniffiLib` (instead of `UniffiLibChecksums`) since it now also performs the contract version check. I think "integrity check" is OK terminology to describe the aggregation of "contract version check" and "checksum checks".

Reply:
@bendk I've ensured to not get lint errors ([which I got at first](https://app.circleci.com/pipelines/github/mozilla/uniffi-rs/5037/workflows/da1d88a7-0392-45bb-8187-9410b1328e5c/jobs/21347)) if `lib: UniFFILib` is not used - which can be the case now if `initialization_fns()` is empty. 

I've updated to only conditionally include the `.also` after loading lib, only if `initialization_fns()` is not empty.

I think that is the solution. Let me know if you are happy with it.